oibackoff/index.d.ts

```diff
@@ -0,0 +1,65 @@
+// Type definitions for oibackoff v1.0
+// Project: https://github.com/chilts/oibackoff
+// Definitions by: Joshua DeVinney <https://github.com/geoffreak>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export declare function backoff(opts: BackoffOptions): BackoffInstance;
+
+export declare interface BackoffOptions {
+    maxTries?: number;
+    algorithm?: 'incremental' | 'exponential' | 'fibonacci';
+    delayRatio?: number;
+    maxDelay?: number;
+}
+
+export declare interface BackoffInstance {
+    <A,B>(fn: (callback: (a: A, b?: B) => any) => any, intermediate: BackoffIntermediate<A>, callback: (a: A, b: (B | null), priorErrors?: A[]) => void): void;
```

Comment:
Callbacks with optional parameters are usually a bad idea. Could this be `priorErrors: A[] | undefined`? (Or just `priorErrors: A[]` if it's always supplied.)